### PR TITLE
bork: Remove `bottle :unneeded` as we're trying to bottle everything

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Related issue: https://github.com/Homebrew/homebrew-core/issues/75943.

### CODE REVIEW

**Comment by Moisan** [comment]
I added the `CI-force-arm` label and restarted the CI, otherwise the ARM bottle isn't built.

**Comment by issyl0** [comment]
Ah, good to know, thanks! I'm clearly a bit out of the loop on core bottling procedures!

**Comment by carlocab** [comment]
```
❯ sha256sum *.tar.gz
9ffc55b10b4adc9759ca4178855eb7f199cf4ba70a2b93d4f57e9131a09cce7c  bork--0.12.0.arm64_big_sur.bottle.tar.gz
6f3c112f62d5a78361f3507bef0fa1d4d229acba674380673491574c4d3294d7  bork--0.12.0.big_sur.bottle.tar.gz
6f3c112f62d5a78361f3507bef0fa1d4d229acba674380673491574c4d3294d7  bork--0.12.0.catalina.bottle.tar.gz
6f3c112f62d5a78361f3507bef0fa1d4d229acba674380673491574c4d3294d7  bork--0.12.0.mojave.bottle.tar.gz
```
Looks like you might need to do some `inreplace`s to make the bottles the same.

<details><summary>diff</summary>
<p>

```
--- bork--0.12.0.arm64_big_sur.bottle.tar.gz
+++ bork--0.12.0.big_sur.bottle.tar.gz
│   --- bork--0.12.0.arm64_big_sur.bottle.tar
├── +++ bork--0.12.0.big_sur.bottle.tar
│ ├── file list
│ │ @@ -101,15 +101,15 @@
│ │  -rwxr-xr-x   0        0        0      951 2021-02-20 12:42:35.000000 bork/0.12.0/test/help-system.bats
│ │  -rwxr-xr-x   0        0        0     1671 2021-02-20 12:42:35.000000 bork/0.12.0/test/help-text.bats
│ │  -rw-r--r--   0        0        0      752 2021-02-20 12:42:35.000000 bork/0.12.0/test/helpers.sh
│ │  -rw-r--r--   0        0        0     1470 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-apk.bats
│ │  -rwxr-xr-x   0        0        0     1549 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-apt.bats
│ │  -rw-r--r--   0        0        0     2056 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-brew-tap.bats
│ │  -rwxr-xr-x   0        0        0     1354 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-brew.bats
│ │ --rwxr-xr-x   0        0        0     1677 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-cask.bats
│ │ +-rwxr-xr-x   0        0        0     1695 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-cask.bats
│ │  -rwxr-xr-x   0        0        0     3351 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-defaults.bats
│ │  -rwxr-xr-x   0        0        0      986 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-directories.bats
│ │  -rw-r--r--   0        0        0     1901 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-directory-perms.bats
│ │  -rw-r--r--   0        0        0     1712 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-download.bats
│ │  -rwxr-xr-x   0        0        0     4935 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-file.bats
│ │  -rwxr-xr-x   0        0        0      717 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-gem.bats
│ │  -rwxr-xr-x   0        0        0     4463 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-git.bats
│ │ @@ -117,26 +117,26 @@
│ │  -rw-r--r--   0        0        0      759 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-go-get.bats
│ │  -rwxr-xr-x   0        0        0      787 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-group.bats
│ │  -rwxr-xr-x   0        0        0      735 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-iptables.bats
│ │  -rwxr-xr-x   0        0        0     1187 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-mas.bats
│ │  -rw-r--r--   0        0        0     1066 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-npm.bats
│ │  -rwxr-xr-x   0        0        0      708 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-pip.bats
│ │  -rwxr-xr-x   0        0        0      723 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-pip3.bats
│ │ --rw-r--r--   0        0        0     8254 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-pipsi.bats
│ │ +-rw-r--r--   0        0        0     8272 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-pipsi.bats
│ │  -rwxr-xr-x   0        0        0     1670 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-symlink.bats
│ │  -rwxr-xr-x   0        0        0     4133 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-user.bats
│ │  -rwxr-xr-x   0        0        0     1357 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-yum.bats
│ │  -rw-r--r--   0        0        0     1790 2021-02-20 12:42:35.000000 bork/0.12.0/test/type-zypper.bats
│ │  drwxr-xr-x   0        0        0        0 2021-02-20 12:42:35.000000 bork/0.12.0/types/
│ │  -rw-r--r--   0        0        0      660 2021-02-20 12:42:35.000000 bork/0.12.0/types/apk.sh
│ │  -rw-r--r--   0        0        0      724 2021-02-20 12:42:35.000000 bork/0.12.0/types/apm.sh
│ │  -rwxr-xr-x   0        0        0     1145 2021-02-20 12:42:35.000000 bork/0.12.0/types/apt.sh
│ │  -rw-r--r--   0        0        0     1342 2021-02-20 12:42:35.000000 bork/0.12.0/types/brew-tap.sh
│ │  -rwxr-xr-x   0        0        0     1769 2021-02-20 12:42:35.000000 bork/0.12.0/types/brew.sh
│ │ --rw-r--r--   0        0        0     1304 2021-02-20 12:42:35.000000 bork/0.12.0/types/cask.sh
│ │ +-rw-r--r--   0        0        0     1313 2021-02-20 12:42:35.000000 bork/0.12.0/types/cask.sh
│ │  -rw-r--r--   0        0        0      359 2021-02-20 12:42:35.000000 bork/0.12.0/types/check.sh
│ │  -rw-r--r--   0        0        0     2537 2021-02-20 12:42:35.000000 bork/0.12.0/types/defaults.sh
│ │  -rw-r--r--   0        0        0     1823 2021-02-20 12:42:35.000000 bork/0.12.0/types/directory.sh
│ │  -rw-r--r--   0        0        0     1124 2021-02-20 12:42:35.000000 bork/0.12.0/types/download.sh
│ │  -rw-r--r--   0        0        0     3169 2021-02-20 12:42:35.000000 bork/0.12.0/types/file.sh
│ │  -rw-r--r--   0        0        0      688 2021-02-20 12:42:35.000000 bork/0.12.0/types/gem.sh
│ │  -rwxr-xr-x   0        0        0     3261 2021-02-20 12:42:35.000000 bork/0.12.0/types/git.sh
│ │ @@ -144,14 +144,14 @@
│ │  -rw-r--r--   0        0        0      411 2021-02-20 12:42:35.000000 bork/0.12.0/types/go-get.sh
│ │  -rw-r--r--   0        0        0      470 2021-02-20 12:42:35.000000 bork/0.12.0/types/group.sh
│ │  -rw-r--r--   0        0        0      675 2021-02-20 12:42:35.000000 bork/0.12.0/types/iptables.sh
│ │  -rw-r--r--   0        0        0      966 2021-02-20 12:42:35.000000 bork/0.12.0/types/mas.sh
│ │  -rw-r--r--   0        0        0      888 2021-02-20 12:42:35.000000 bork/0.12.0/types/npm.sh
│ │  -rw-r--r--   0        0        0      454 2021-02-20 12:42:35.000000 bork/0.12.0/types/pip.sh
│ │  -rw-r--r--   0        0        0      440 2021-02-20 12:42:35.000000 bork/0.12.0/types/pip3.sh
│ │ --rw-r--r--   0        0        0     3699 2021-02-20 12:42:35.000000 bork/0.12.0/types/pipsi.sh
│ │ +-rw-r--r--   0        0        0     3717 2021-02-20 12:42:35.000000 bork/0.12.0/types/pipsi.sh
│ │  -rw-r--r--   0        0        0      520 2021-02-20 12:42:35.000000 bork/0.12.0/types/scutil.sh
│ │ --rw-r--r--   0        0        0      390 2021-02-20 12:42:35.000000 bork/0.12.0/types/shells.sh
│ │ +-rw-r--r--   0        0        0      399 2021-02-20 12:42:35.000000 bork/0.12.0/types/shells.sh
│ │  -rw-r--r--   0        0        0      748 2021-02-20 12:42:35.000000 bork/0.12.0/types/symlink.sh
│ │  -rw-r--r--   0        0        0     2520 2021-02-20 12:42:35.000000 bork/0.12.0/types/user.sh
│ │  -rwxr-xr-x   0        0        0      621 2021-02-20 12:42:35.000000 bork/0.12.0/types/yum.sh
│ │  -rw-r--r--   0        0        0      770 2021-02-20 12:42:35.000000 bork/0.12.0/types/zypper.sh
│ ├── bork/0.12.0/test/type-cask.bats
│ │ @@ -1,15 +1,15 @@
│ │  #!/usr/bin/env bats
│ │  
│ │  . test/helpers.sh
│ │  cask () { . $BORK_SOURCE_DIR/types/cask.sh $*; }
│ │  
│ │  setup () {
│ │    respond_to "uname -s" "echo Darwin"
│ │ -  respond_to "which brew" "echo /usr/local/bin/brew"
│ │ +  respond_to "which brew" "echo @@HOMEBREW_PREFIX@@/bin/brew"
│ │    respond_to "brew list --cask" "cat $fixtures/cask-list.txt"
│ │  }
│ │  
│ │  @test "cask status reports unsupported platforms" {
│ │    respond_to "uname -s" "echo Linux"
│ │    run cask status something
│ │    [ "$status" -eq $STATUS_UNSUPPORTED_PLATFORM ]
│ │ @@ -50,10 +50,10 @@
│ │    [ "$output" = 'brew install --cask missing_package' ]
│ │  }
│ │  
│ │  @test "cask upgrade performs a force install and cleans up old versions" {
│ │    run cask upgrade installed_package
│ │    [ "$status" -eq 0 ]
│ │    run baked_output
│ │ -  [ "${lines[0]}" = "rm -rf /usr/local/Caskroom/installed_package" ]
│ │ +  [ "${lines[0]}" = "rm -rf @@HOMEBREW_PREFIX@@/Caskroom/installed_package" ]
│ │    [ "${lines[1]}" = "brew install --cask installed_package --force" ]
│ │  }
│ ├── bork/0.12.0/test/type-pipsi.bats
│ │ @@ -1,13 +1,13 @@
│ │  #!/usr/bin/env bats
│ │  
│ │  . test/helpers.sh
│ │  
│ │ -pipsi_global_bin_dir="/usr/local/bin"
│ │ -pipsi_global_home="/usr/local/lib/pipsi"
│ │ +pipsi_global_bin_dir="@@HOMEBREW_PREFIX@@/bin"
│ │ +pipsi_global_home="@@HOMEBREW_PREFIX@@/lib/pipsi"
│ │  
│ │  pipsi() { . $BORK_SOURCE_DIR/types/pipsi.sh $*; }
│ │  
│ │  setup() {
│ │    respond_to 'pipsi list' "cat $fixtures/pipsi-list.txt"
│ │  }
│ ├── bork/0.12.0/types/cask.sh
│ │ @@ -33,15 +33,15 @@
│ │      else
│ │        bake brew install --cask $name
│ │      fi
│ │      ;;
│ │  
│ │    upgrade)
│ │      # TODO move rm statement to remove action with clobber
│ │ -    bake rm -rf "/usr/local/Caskroom/$name"
│ │ +    bake rm -rf "@@HOMEBREW_PREFIX@@/Caskroom/$name"
│ │      if [ -n "$appdir" ]; then
│ │        bake brew install --cask $name --appdir=$appdir --force
│ │      else
│ │        bake brew install --cask $name --force
│ │      fi
│ │      ;;
│ ├── bork/0.12.0/types/pipsi.sh
│ │ @@ -19,16 +19,16 @@
│ │  
│ │  # set paths, try variables pipsi itself uses first and fall-back to
│ │  # defaults; take the 'global' option into account
│ │  declare -a pipsi_opts=( )
│ │  pipsi_bin_dir="${PIPSI_BIN_DIR:-"${HOME}/.local/bin"}"
│ │  pipsi_home="${PIPSI_HOME:-"${HOME}/.local/venvs"}"
│ │  if [[ ${global} == true ]]; then
│ │ -  pipsi_bin_dir="${PIPSI_GLOBAL_BIN_DIR:-"/usr/local/bin"}"
│ │ -  pipsi_home="${PIPSI_GLOBAL_HOME:-"/usr/local/lib/pipsi"}"
│ │ +  pipsi_bin_dir="${PIPSI_GLOBAL_BIN_DIR:-"@@HOMEBREW_PREFIX@@/bin"}"
│ │ +  pipsi_home="${PIPSI_GLOBAL_HOME:-"@@HOMEBREW_PREFIX@@/lib/pipsi"}"
│ │    pipsi_opts+=( --bin-dir=${pipsi_bin_dir} --home=${pipsi_home} )
│ │  fi
│ │  pip="${pipsi_home}/${name}/bin/pip"
│ │  
│ │  # if bin_dir or home are not writable, automatically attempt to elevate
│ │  # permissions using sudo
│ │  get_su() {
│ ├── bork/0.12.0/types/shells.sh
│ │ @@ -1,15 +1,15 @@
│ │  action=$1
│ │  shell=$2
│ │  shift 2
│ │  
│ │  case $action in
│ │  	desc)
│ │  		echo "asserts presence of a shell in /etc/shells"
│ │ -		echo "> shells /usr/local/bin/zsh"
│ │ +		echo "> shells @@HOMEBREW_PREFIX@@/bin/zsh"
│ │  	;;
│ │      status)
│ │          bake cat /etc/shells | grep "$shell"
│ │          [ "$?" -gt 0 ] && return $STATUS_MISSING
│ │          return $STATUS_OK
│ │      ;;
```

</p>
</details>

Not sure what to do about the `Caskroom` reference, though, since that actually shouldn't be relocated.

File: Formula/bork.rb [hunk 1]
```diff
@@ -6,9 +6,11 @@ class Bork < Formula
   license "Apache-2.0"
   head "https://github.com/skylarmacdonald/bork.git", branch: "main"
 
-  bottle :unneeded
-
   def install
+    %w[types/shells.sh types/pipsi.sh types/cask.sh test/type-pipsi.bats test/type-cask.bats].each do |f|
+      inreplace f, "/usr/local/", HOMEBREW_PREFIX
+    end
```
Comment (carlocab) [inline Formula/bork.rb:12]:
```suggestion
    files = %w[types/shells.sh types/pipsi.sh types/cask.sh test/type-pipsi.bats test/type-cask.bats]
    inreplace files, "/usr/local/", HOMEBREW_PREFIX
```
I think `inreplace` can take an array as its first argument. Though

**Review by carlocab — COMMENTED** [review]
~~I think `inreplace` can take an array as its first argument.~~ oops, posted this twice

File: Formula/bork.rb (hunk 1 above)
Comment (issyl0) [reply Formula/bork.rb:12]: Cool, thanks Carlo! If this `inreplace` solves the "ARM not identical to Intel" problem then I'll apply these syntax changes afterwards.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**